packages/vite/src/node/optimizer/index.ts

```diff
@@ -624,23 +581,66 @@ export function runOptimizeDeps(
           }
         }
 
-        const dataPath = path.join(processingCacheDir, '_metadata.json')
-        writeFile(
-          dataPath,
-          stringifyDepsOptimizerMetadata(metadata, depsCacheDir),
-        )
-
         debug(
           `Dependencies bundled in ${(performance.now() - start).toFixed(2)}ms`,
         )
 
-        return createProcessingResult()
+        return {
+          metadata,
+          async commit() {
+            // Write this run of pre-bundled dependencies to the deps cache
+
+            let oldFilesNames: string[]
+            if (!fs.existsSync(depsCacheDir)) {
+              fs.mkdirSync(depsCacheDir, { recursive: true })
+              oldFilesNames = []
+            } else {
+              oldFilesNames = await fsp.readdir(depsCacheDir)
+            }
+
+            const files = []
+
+            // a hint for Node.js
+            // all files in the cache directory should be recognized as ES modules
+            const packageJsonFilePath = path.resolve(
+              depsCacheDir,
+              'package.json',
+            )
+            if (!fs.existsSync(packageJsonFilePath)) {
+              files.push(
+                fsp.writeFile(
+                  packageJsonFilePath,
+                  '{\n  "type": "module"\n}\n',
+                ),
+              )
+            }
+
+            const dataPath = path.join(depsCacheDir, '_metadata.json')
+            files.push(
+              fsp.writeFile(
+                dataPath,
+                stringifyDepsOptimizerMetadata(metadata, depsCacheDir),
+              ),
+            )
+
+            for (const outputFile of result.outputFiles!) {
+              files.push(fsp.writeFile(outputFile.path, outputFile.text))
+            }
+
+            // Clean up old files in the background
+            cleanupDepsCacheStaleFiles(oldFilesNames, depsCacheDir, metadata)
```

Comment:
We could perhaps fast-path this by not cleaning up if the deps cache dir previously doesn't exist. Since we're already creating a new empty folder

Reply:
`oldFileNames` would be empty in that case, so this is already happening, not?